syft/source/directory_resolver.go

```diff
@@ -207,6 +216,7 @@ func (r directoryResolver) FilesByPath(userPaths ...string) ([]Location, error)
 			log.Warnf("unable to get file by path=%q : %+v", userPath, err)
 			continue
 		}
+		// TODO: why not use stored metadata?
```

Comment:
note: I left this as a question for future us, not necessarily one to solve on this PR